### PR TITLE
Favourite games

### PR DESCRIPTION
**Pull request by amirine**

### CODE REVIEW

File: game/models.py [hunk 1]
```diff
@@ -1,3 +1,32 @@
...
+from game.igdb_wrapper import IGDBRequestsHandler
 
-# Create your models here.
+
+class Game(models.Model):
+    """Game model"""
+
+    game_id = models.IntegerField()
+
+    def get_game_info_musts(self):
```
Comment (Satevg) [inline game/models.py]: nitpick: better to name this function `get_game_info_by_id`

File: game/views.py [hunk 2]
```diff
@@ -123,8 +133,63 @@ def get(request, game_id):
             'tweets': tweets.request_return_handle(game['name'], DetailPageView.TWEETS_LIMIT),
         }
 
+        if request.user.is_authenticated:
+            context.update({
+                'is_added': game_id in list(request.user.favourites.values_list('game__game_id', flat=True))
```
Comment (Satevg) [inline game/views.py]: nitpick: Can be queried directly from `UserFavouriteGame` model.

btw seems that check for `is_deleted` is missing here

File: game/views.py [hunk 3]
```diff
@@ -123,8 +133,63 @@ def get(request, game_id):
...
+        """
+        Adds game to favourites on must button click, deletes game from favourites on unmust button click.
+        Works on both main and musts pages must buttons.
+        """
+
+        if game_id not in list(request.user.favourites.values_list('game__game_id', flat=True)):
+            current_game, created = Game.objects.get_or_create(game_id=game_id)
+            request.user.favourites.create(game=current_game, user=request.user)
+        else:
+            request.user.favourites.get(game__game_id=game_id).delete()
```
Comment (Satevg) [inline game/views.py]: I think that it's needed just set `is_deleted=True`

File: game/views.py [hunk 4]
```diff
@@ -123,8 +133,63 @@ def get(request, game_id):
...
+    View for musts page: displays favourite user games with the opportunity to soft delete items.
+    Soft delete and add performed by button unmust/must click.
+    """
+
+    GAMES_LIMIT = 18
+    GAMES_PER_PAGE = 12
+
+    @staticmethod
+    def get(request):
+        """Gets favourite user games (soft deleted items are deleted completely) """
```
Comment (Satevg) [inline game/views.py]: yeah, no need to delete them. They should be still in database, if entry already created then it's needed tick/untick is_deleted flag

**Review by Satevg — CHANGES_REQUESTED** [review]

File: game/models.py (hunk 1 above)
Comment (amirine) [reply game/models.py]: Done👌

File: game/views.py (hunk 4 above)
Comment (amirine) [reply game/views.py]: Have fixed soft delete functionality on all pages, soft deleted items are stored in the database now👍

**Review by Satevg — APPROVED** [review]

**Pull request merged**